cli/pcluster/config/config_patch.py

```diff
@@ -0,0 +1,256 @@
+                target_section = self.target_config.get_section(section_key, section_label)
+                base_section = self._get_config_section(self.base_config, target_section)
+                base_section.visited = True
+                self._compare_section(base_section, target_section)
+
+        # Then, compare all non visited base sections vs target config.
+        for section_key in sorted(self.base_config.get_section_keys()):
+            for section_label in sorted(self.base_config.get_sections(section_key).keys()):
+                base_section = self.base_config.get_section(section_key, section_label)
+                if not hasattr(base_section, "visited"):
+                    target_section = self._get_config_section(self.target_config, base_section)
+                    self._compare_section(base_section, target_section)
+
+    def _compare_section(self, base_section, target_section):
+        """
+        Compare the provided base and target sections and append the detected changes to the internal changes list.
+
+        :param base_section: The section in the base configuration
+        :param target_section: The corresponding section in the target configuration
+        """
+        for _, param in target_section.params.items():
+            base_value = base_section.get_param_value(param.key)
+            target_value = param.value
+
+            if not self._compare_params(base_section.key, param.key, base_value, target_value):
+                self.changes.append(
+                    Change(
+                        target_section.key,
+                        target_section.label,
+                        param.key,
+                        base_value,
+                        param.value,
+                        param.get_update_policy(),
+                    )
+                )
+
+    def _compare_params(self, section_key, param_key, base_value, target_value):
+        """
+        Compare the specified base configuration parameter with the corresponding one in target configuration.
+
+        The way the comparison is performed depends on the availability of a custom comparator for the specified
+        parameter. If no custom comparator is available, a default equality comparison is performed.
+
+        :param section_key: The section key
+        :param param_key: The parameter key
+        :param base_value: The value of the parameter in the base configuration
+        :param target_value: The value of the parameter in the target configuration
+        :return: True if the values can be considered equal, False otherwise
+        """
+        comparator = ConfigPatch.PARAM_COMPARATORS.get((section_key.lower(), param_key.lower()))
+        if comparator:
+            result = comparator(base_value, target_value)
+        else:
+            result = base_value == target_value
+        return result
+
+    def _remove_ignored_sections(self, config):
```

Comment:
We could also avoid this method by adding a new attribute in the mappings file to threat the aws, global and aliases sections as special sections (e.g. we could add an attribute `"global": True`) in this way we don't need to have another list here to store the ignored sections.

We have the same list in the tests to so in this way we can rely on the mappings.py instead of having this static list.
What do you think?

Reply:
What about returning only the internal sections from PClusterConfig, so that global sections, as expected, won't be seen externally?